cargo-dist/templates/ci/partials/publish_npm.yml.j2

```diff
@@ -0,0 +1,24 @@
+  publish-npm:
+    needs:
+      - plan
+      - host
+    {{%- for job in host_jobs %}}
+      - custom-{{{ job|safe }}}
+    {{%- endfor %}}
+    runs-on: {{{ global_task.runner }}}
+    env:
+      NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+      PLAN: ${{ needs.plan.outputs.val }}
+    if: ${{ !fromJson(needs.plan.outputs.val).announcement_is_prerelease || fromJson(needs.plan.outputs.val).publish_prereleases }}
+    steps:
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+          registry-url: 'https://registry.npmjs.org'
+      - run: |
+          for release in $(echo "$PLAN" | jq --compact-output '.releases[] | select([.artifacts[] | endswith("-npm-package.tar.gz")] | any)'); do
+            pkg=$(echo "$release" | jq '.artifacts[] | select(endswith("-npm-package.tar.gz"))' --raw-output)
+            npm publish --access public pkg
```

Comment:
And whichever dir your fetch the artifacts to, you need to add that dir as a prefix to "pkg" here